### PR TITLE
feat: readonly global config

### PR DESCRIPTION
**Pull request by lkingland**
- :gift:  adds initial support for a global config in `~/.config/func/config.yaml`
- :gift: default for `func create --language` can now be defined in global config
- :gift: defaulting to confirm prompts can be enabled in the global config
- :gift: modifed defaults are visible in help text.

Example  `~/.config/func/config.yaml`:
```
language: go
confirm: true
```

The config location is mutable using `XDG_CONFIG_HOME`.  A config file is not required for normal operation.  Values are accurately reflected in command help text.

This is the first part of a general-purpose Global Config (#901) which is prerequisite to, and will also partially  address the interrelated issues #573 #636 and #1097. 

/kind enhancement

### CODE REVIEW

File: config/config.go
Comment (lance) [inline config/config.go:59]:
```suggestion
    cfg, err = Load(p)
    if os.IsNotExist(err) {
		err = nil // config file is not required
	}
```

**Review by lance — COMMENTED** [review]
/lgtm with one small suggestion
/hold for others

File: config/config.go
Comment (lkingland) [reply config/config.go:59]: /facepalm
Pretty sure that's the reason I added the `Load` method in the first place.
_codus interruptus_ perhaps!

Comment (lkingland) [reply config/config.go]: Hey @lance; while in half-sleep late last night I realized why this was the way it was.
The Config object is first instantiated with its values populated with static defaults.
Then the .yaml is loaded into that sams struct, overwriting any static defaults with user-provided global settings.
The `Load` method merely loads exactly the .yaml, ignoring static defaults.
Therefore, updating this method in this manner will ignore the `cfg` object and thus static defaults
(hence the code staticcheck complaining about the unused declaration).
This is the general path: static defaults first, user provided config values overwrite those, environment variables overwrite those, and finally flags.  👍🏻  

I refactored a bit to make this more clear so as to hopefully avoid the same confusion in the future.

**Comment by lkingland** [comment]
> This branch has conflicts that must be resolved
> Use the [web editor](https://github.com/knative-sandbox/kn-plugin-func/pull/1260/conflicts) or the to resolve conflicts.
> Conflicting files
> docker/creds/credentials_test.go


Screen Shot 2022-09-22 at 11 23 01 PM

File: config/config.go
Comment (salaboy) [inline config/config.go:31]: @lkingland so the idea here is to add for example a global default registry for all functions? I mean in another PR

Comment (lkingland) [reply config/config.go:31]: Yes, along with any other settings who would benefit from a shared configurable default.

**Review by lance — COMMENTED** [review]
/lgtm

File: config/config.go
Comment (zroubalik) [reply config/config.go:31]: 💯

**Review by zroubalik — APPROVED** [review]
/lgtm

great job @lkingland !

**Comment by lance** [comment]
Thanks for allowing commits from maintainers @lkingland - I have resolved the merge conflicts. Assuming tests pass, looks good to me.

/lgtm

**Pull request merged**
